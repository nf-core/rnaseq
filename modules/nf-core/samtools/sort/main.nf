process SAMTOOLS_SORT {
    tag "$meta.id"
    label 'process_medium'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/samtools:1.19.2--h50ea8bc_0' :
        'biocontainers/samtools:1.19.2--h50ea8bc_0' }"

    input:
    tuple val(meta), path(bam)

    output:
    tuple val(meta), path("*.bam"), emit: bam
    tuple val(meta), path("*.csi"), emit: csi, optional: true
    path  "versions.yml"          , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    if ("$bam" == "${prefix}.bam") error "Input and output names are the same, use \"task.ext.prefix\" to disambiguate!"
    """
    samtools sort \\
        $args \\
        -@ $task.cpus \\
        -o ${prefix}.bam \\
        -T $prefix \\
        $bam

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        samtools: \$(echo \$(samtools --version 2>&1) | sed 's/^.*samtools //; s/Using.*\$//')
    END_VERSIONS
    """

    stub:
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}.bam

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        samtools: \$(echo \$(samtools --version 2>&1) | sed 's/^.*samtools //; s/Using.*\$//')
    END_VERSIONS
    """
}
