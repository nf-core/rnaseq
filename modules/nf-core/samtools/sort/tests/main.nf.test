nextflow_process {

    name "Test Process SAMTOOLS_SORT"
    script "../main.nf"
    process "SAMTOOLS_SORT"
    tag "modules"
    tag "modules_nfcore"
    tag "samtools"
    tag "samtools/sort"

    test("bam") {

        config "./nextflow.config"

        when {
            params {
                outdir   = "$outputDir"
            }
            process {
                """
                input[0] = Channel.of([
                    [ id:'test', single_end:false ], // meta map
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/bam/test.paired_end.bam', checkIfExists: true)
                ])
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("bam_stub") {

        config "./nextflow.config"
        options "-stub-run"

        when {
            params {
                outdir   = "$outputDir"
            }
            process {
                """
                input[0] = Channel.of([
                    [ id:'test', single_end:false ], // meta map
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/bam/test.paired_end.bam', checkIfExists: true)
                ])
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(file(process.out.bam[0][1]).name).match("bam_stub_bam") },
                { assert snapshot(process.out.versions).match("bam_stub_versions") }
            )
        }
    }
}
