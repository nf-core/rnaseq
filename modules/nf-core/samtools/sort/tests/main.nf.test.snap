{
    "bam_stub_bam": {
        "content": [
            "test.sorted.bam"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.04.3"
        },
        "timestamp": "2024-02-12T19:21:04.364044"
    },
    "bam_stub_versions": {
        "content": [
            [
                "versions.yml:md5,e6d43fefc9a8bff91c2ce6e3a1716eca"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.01.0"
        },
        "timestamp": "2024-02-13T16:15:00.20800281"
    },
    "bam": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.sorted.bam:md5,c6ea1346ec4aae007eb40b708935088c"
                    ]
                ],
                "1": [
                    
                ],
                "2": [
                    "versions.yml:md5,e6d43fefc9a8bff91c2ce6e3a1716eca"
                ],
                "bam": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.sorted.bam:md5,c6ea1346ec4aae007eb40b708935088c"
                    ]
                ],
                "csi": [
                    
                ],
                "versions": [
                    "versions.yml:md5,e6d43fefc9a8bff91c2ce6e3a1716eca"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.01.0"
        },
        "timestamp": "2024-02-13T16:14:52.736359271"
    }
}